java/flight/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -90,20 +86,86 @@ public void close() throws InterruptedException {
     }
   }
 
-  public interface OutputFlight {
-    void sendData(int count);
+  public static Builder builder(BufferAllocator allocator, Location location, FlightProducer producer) {
+    return new Builder(allocator, location, producer);
+  }
 
-    void done();
+  public static final class Builder {
 
-    void fail(Throwable t);
-  }
+    private final BufferAllocator allocator;
+    private final Location location;
+    private final FlightProducer producer;
+    private ServerAuthHandler authHandler = ServerAuthHandler.NO_OP;
 
-  public interface FlightServerHandler {
+    private InputStream certChain;
+    private InputStream key;
 
-    public FlightInfo getFlightInfo(String descriptor) throws Exception;
+    Builder(BufferAllocator allocator, Location location, FlightProducer producer) {
+      this.allocator = allocator;
+      this.location = location;
+      this.producer = producer;
+    }
 
-    public OutputFlight setupFlight(VectorSchemaRoot root);
+    public FlightServer build() {
+      final NettyServerBuilder builder;
+      switch (location.getUri().getScheme()) {
+        case LocationSchemes.GRPC_DOMAIN_SOCKET: {
+          // TODO: need reflection to check if domain sockets are available
+          throw new UnsupportedOperationException("Domain sockets are not available.");
```

Comment:
Can we include support for domain socket in this patch? It would be nice to confirm that they also work cross platform.

Reply:
For gRPC/Netty, they are Unix-only and require a Unix-only dependency. If that's OK, I'll add tests for that.

Reply:
Let's add this and only fail if we're running some place else.